AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m

```diff
@@ -89,7 +92,22 @@ - (void)setUserId:(nullable NSString *)userId {
     [MS_USER_DEFAULTS setObject:[NSKeyedArchiver archivedDataWithRootObject:self.userIdHistory] forKey:kMSUserIdHistoryKey];
     MSLogVerbose([MSAppCenter logTag], @"Stored new userId:%@ and timestamp: %@.", self.currentUserIdInfo.userId,
                  self.currentUserIdInfo.timestamp);
+    if (![self changeUserIdIfDifferent:self.mUserId]){
+      return;
+    }
+    synchronizedDelegates = [self.delegates allObjects];
+    for (id<MSUserIdContextDelegate> delegate in synchronizedDelegates) {
+      [delegate onNewUserId:self];
+    }
+  }
+}
+
+- (BOOL) changeUserIdIfDifferent:(NSString *)userId {
+  if (self.mUserId != nil && [self.mUserId isEqualToString:userId]) {
```

Comment:
It is incorrect. Fix the `nil == nil` case

Reply:
This one is removed; Resolving